### DISCUSSION TITLE
SQL foreign key contraint is incorrectly formed

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#4839] [Help] [answered]
Hello,

I'm trying to add a relationship between two MySQL tables, in this case One (organisations) to Many (fiches).
I get this error : 
Error saving column
[Builder] HTTP 500 on POST:https://***/api/tables
Can't create table `xxx_prod`.`fiches` (errno: 150 "Foreign key constraint is incorrectly formed")

I tried via the 'Create column' button on the 'fiches' or 'organisations'.

After this the column 'fk_fiches_organisations' is added to the 'fiches' in the sql table but no foreign key in the sql table or column in Budibase.
I can try again but I'm getting this error:
message	"Duplicate column name 'fk_fiches_organisations'"
status	500

I have to manually delete the column in sql to get back to the initial situation.

Thank you !

**Comment by mjashanks** [comment]
Hey @diogenesbrussels - I have tried this one in MySQL and it is working as expected.

Are you able to provide
- Your MySQL version
- `CREATE` scripts for both tables (don't need the data). If you don't want to post it here, feel free to email to support@budibase.com

**Comment by diogenesbrussels** [comment, marked as answer]
Hello,

We are using 10.3.31-MariaDB-0+deb10u1 Debian 10, I've just emailed you the SQL commands.

Thank you!

**Reply by mjashanks** [reply]
Ok, I found the issue.

The reason this is happening is because the primary key on your `organistations` table is not unique.

So, I
1. checked the `UN` box
![image](https://user-images.githubusercontent.com/3524181/158366719-947af41d-8769-4410-9c90-7d4b25e0682c.png)
2.   applied changes
3. removed the `fk_organisations_fiches` column in the other table
4. refreshed my Budibase browser tab 
5. created the relationship successfully

We need to document this better of course. 

Thanks for pointing it out!

**Reply by diogenesbrussels** [reply]
Great thank you ! 
I think it is linked to the UNSIGNED property and not unique property (because it is already a primary key and therefore unique).
It works now, maybe when fetching tables Budibase could display a warning for non-compliant tables/fields ?
